src/purifier.js

```diff
@@ -276,6 +280,18 @@ export function addPurifyHooks(purifier, diffing) {
     });
     allowedAttributesChanges.length = 0;
 
+    // Remove input type file if applicable. The purifier will actually
+    // not remove this attribute because of a Safari bug where removing it
+    // will result in not being able to add it programmatically afterwards.
+    // For AMP HTML's usage, this is fine.
+    const tagName = node.nodeName.toLowerCase();
+    if (tagName == 'input') {
+      const inputType = node.getAttribute('type');
+      if (inputType == 'file' && isAmp4Email(doc)) {
+        node.removeAttribute('type');
+      }
+    }
```

Comment:
I don't quite follow. Let's chat when you have a minute?

Reply:
> I modified purifier to manually remove the input type file attribute as it is not allowed by DOM Purify. 

I don't think this should be an issue. For AMP, `[type=file]` attribute won't be removed per the DOMPurify line. For AMP4EMAIL, it will be removed but won't be re-added.

Reply:
Yes, let's chat. Without this change, Dom Purify will not remove the [type='file'] attribute, as the following test would fail if the amp format type is AMP4Email.

    purify('<input type='file'>').to.equal('<input>');